src/WindowsTracing/KrabsTracer.cpp

```diff
@@ -43,11 +50,45 @@ void KrabsTracer::EnableProviders() {
   context_switch_provider_.add_on_event_callback(
       [this](const auto& record, const auto& context) { OnThreadEvent(record, context); });
   trace_.enable(context_switch_provider_);
+
+  stack_walk_provider_.add_on_event_callback(
+      [this](const auto& record, const auto& context) { OnStackWalkEvent(record, context); });
+  trace_.enable(stack_walk_provider_);
+}
+
+void KrabsTracer::EnableSystemProfilePrivilege(bool value) {
+  auto result = AdjustTokenPrivilege(SE_SYSTEM_PROFILE_NAME, value);
+  if (result.has_error()) ERROR("%s", result.error().message());
+}
+
+void KrabsTracer::SetupStackTracing() {
+  // Set sampling frequency for ETW trace. Note that the session handle must be 0.
+  const double frequency = capture_options_.samples_per_second();
+  CHECK(frequency > 0);
```

Comment:
Remember that we can actually disable callstack sampling in the capture options completely. We should not crash in this case, but for sure you can ensure to not call this method in this case.

Reply:
A frequency of 0 is used in that case right? I added an early out if the frequency is 0, and a check that the frequency is not a negative value.